core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import alluxio.client.file.URIStatus;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Get bucket result defined in http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html
+ * It will be encoded into an XML string to be returned as an error response for the REST call.
+ *
+ * TODO(chaomin): consider add more required fields in S3 BucketGet API.
+ */
+@JacksonXmlRootElement(localName = "ListBucketResult")
+@JsonPropertyOrder(
+    { "Name", "Prefix", "KeyCount", "MaxKeys", "IsTruncated", "Contents" })
+public class ListBucketResult {
+  // Name of the bucket.
+  private String mName = "";
+  // Keys that begin with the indicated prefix.
+  private String mPrefix = null;
+  // Returns the number of keys included in the response. The value is always less than or equal
+  // to the mMaxKeys value.
+  private int mKeyCount = 0;
+  // The maximum number of keys returned in the response body.
+  private int mMaxKeys = 0;
+  // Specifies whether or not all of the results were returned. If the number of
+  // results exceeds that specified by MaxKeys, all of the results might not be returned.
+  // TODO(chaomin): set mIsTruncated once Alluxio supports maxKeys and partial result in response.
```

Comment:
yes, I agree we should support `maxKeys` for GET bucket operation.
We will need to maintain some state of the current key, since Alluxio FS does not support such op natively. I'd prefer to do this in a separate PR because it is not trivial.

So with this PR, GET bucket always return the full list.